src/api/db/connect.ts

```diff
@@ -1,34 +1,32 @@
 import mongoose from 'mongoose';
 import { InternalServerError } from '../errors.js';
+import { Config } from '../../config/config.js';
 
 // TODO: consider using multiple connections (one per model)
 // to reduce risk of slow trains
 // https://mongoosejs.com/docs/connections.html#multiple_connections
 // https://thecodebarbarian.com/slow-trains-in-mongodb-and-nodejs
 
-let cachedConnectionPromise = null;
+let client: typeof mongoose | null = null;
 
-async function connectToDatabase(config) {
-  if (!cachedConnectionPromise) {
+async function connectToDatabase(config: Config) {
+  if (!client) {
     // If no connection promise is cached, create a new one.
     // We cache the promise instead of the connection itself to prevent race
     // conditions where connect is called more than once.
     const uri = config['/DB/MONGO_DB_URL'];
-    cachedConnectionPromise = await mongoose.connect(uri, {
+    client = await mongoose.connect(uri, {
       bufferCommands: false,
       // TODO: double check auto indexing is off in prod environment
-      autoIndex: process.env.STAGE !== 'prod'
+      autoIndex: process.env.STAGE !== 'prod',
     });
   }
 
   try {
-    const client = await cachedConnectionPromise;
     return client;
```

Comment:
@alukach are you sure don't we still have to await this?

Reply:
According to the TS types, `mongoos.connect()` returns a promise of a client. Once we await that, there's no reason to await it again.  However, we never hit a bug because you can await non-promises as well:

```js
> async function foo() {
... const value = await 99;
... return value;
... }
undefined
> foo().then(console.log)
Promise {
  <pending>,
  [Symbol(async_id_symbol)]: 431,
  [Symbol(trigger_async_id_symbol)]: 428
}
> 99
```

Reply:
ah, ok makes sense!